wdl2cwl/main.py

```diff
@@ -107,6 +107,34 @@ def get_ram_min_js(ram_min: str, unit: str) -> str:
     return js_str
 
 
+def get_ram_min_js_from_function(ram_min_list: List[str], unit: str) -> str:
+    """Get memory requirement for user input from select_first function."""
+    inputs_list = [inputs(input_name) for input_name in ram_min_list]
+    js_str = (
+        '${\nvar unit = "'
+        + unit
+        + '";\n'
+        + "var select_first = (function() {for (const elem of ["
+        + ",".join(inputs_list)
+        + "]) if (elem != null) return elem;}) ();\n"
+        + 'if (select_first == undefined) throw "error! array contains only null values or is empty"\n'
+        + "var value = parseInt(select_first.match(/[0-9]+/g));\n"
+        + 'var memory = "";\n'
+        + 'if(unit==="KiB") memory = value/1024;\n'
+        + 'else if(unit==="MiB") memory = value;\n'
+        + 'else if(unit==="GiB") memory = value*1024;\n'
+        + 'else if(unit==="TiB") memory = value*1024*1024;\n'
+        + 'else if(unit==="B") memory = value/(1024*1024);\n'
+        + 'else if(unit==="KB" || unit==="K") memory = (value*1000)/(1024*1024);\n'
+        + 'else if(unit==="MB" || unit==="M") memory = (value*(1000*1000))/(1024*1024);\n'
+        + 'else if(unit==="GB" || unit==="G") memory = (value*(1000*1000*1000))/(1024*1024);\n'
+        + 'else if(unit==="TB" || unit==="T") memory = (value*(1000*1000*1000*1000))/(1024*1024);\n'
+        + "return parseInt(memory);\n}"
+    )
```

Comment:
Please refactor this function and `get_ram_min_js_from_reference` so there is less code duplication